pkg/controller/sparkapplication/submission.go

```diff
@@ -96,7 +96,7 @@ func buildSubmissionCommandArgs(app *v1beta1.SparkApplication, submissionID stri
 	args = append(args, "--deploy-mode", string(app.Spec.Mode))
 	args = append(args, "--conf", fmt.Sprintf("spark.kubernetes.namespace=%s", app.Namespace))
 	args = append(args, "--conf", fmt.Sprintf("spark.app.name=%s", app.Name))
-	args = append(args, "--conf", fmt.Sprintf("spark.kubernetes.driver.pod.name=%s", getDefaultDriverPodName(app)))
+	args = append(args, "--conf", fmt.Sprintf("spark.kubernetes.driver.pod.name=%s", driverPodName))
```

Comment:
we need to remove driver pod name conf explicitly while adding the sparkConfig  to the args as well as in the `addDriverConfOptions` method. I had something like  https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/495/files#diff-8c1e9e387266f24561c52caf907d1793 in mind

Reply:
Also lets use `config.SparkDriverPodNameKey` for the key.

Reply:
Done.

Reply:
Please also remove the same from `addDriverConfOptions`  method. lines 261-264 in this file.

Reply:
Done